src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -1574,10 +1647,12 @@ mgp_error mgp_result_record_insert(mgp_result_record *record, const char *field_
     if (find_it == record->signature->end()) {
       throw std::out_of_range{fmt::format("The result doesn't have any field named '{}'.", field_name)};
     }
+
```

Comment:
Where is logic for this insertion of values if not deleted for regular procedure in c and c++?

Reply:
The deletion check cannot be done inside `mgp_result_record_insert` because, upon trying to insert a deleted value, the entire record should be skipped. This means that people writing query modules using the C and C++ APIs need to do a check manually.